src/App.tsx

```diff
@@ -15,37 +15,94 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT = 'DEFAULT',
+  ALPHABETICAL = 'ALPHABETICAL',
+  LENGTH = 'LENGTH',
+}
 export const App: React.FC = () => {
+  const initialGoods = [...goodsFromServer];
+  const [goods, setGoods] = useState<string[]>(initialGoods);
+  const [sortType, setSortType] = useState<SortType>(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortGoods = (type: SortType) => {
+    let sortedGoods = [...initialGoods];
+
+    switch (type) {
+      case SortType.ALPHABETICAL:
+        sortedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case SortType.LENGTH:
+        sortedGoods.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        sortedGoods = initialGoods;
+        break;
+    }
+
+    if (isReversed) {
+      sortedGoods.reverse();
+    }
+
+    setGoods(sortedGoods);
+    setSortType(type);
+  };
+
+  const toggleReverse = () => {
+    setGoods(prevGoods => [...prevGoods].reverse());
+    setIsReversed(!isReversed);
+  };
+
+  const resetGoods = () => {
+    setGoods([...initialGoods]);
```

Comment:
Ensure that the `resetGoods` function correctly resets the `goods` state to the initial order. This line seems correct, but verify that it works as expected in all scenarios, especially after multiple sorts and reverses.